### ISSUE TITLE
ERROR: import declarations may only appear at top level of a module

### ISSUE THREAD

**Issue by MissingUser** [MichaelAquilina/improved-workspace-indicator#36]
I have downloaded the last version for gnome 45 and followed  the install instructions without any compile or install warning.
After restartig  extension is not working with error message: "import declarations may only appear at top level of a module".

My system:
Ubuntu 23.10 x86_64 mantic
Kernel 6.5.0-10-generic
GNOME Shell 45.0
Running on Wayland

**Comment by MichaelAquilina** [comment]
Yep unfortunately gnome 45 comes with breaking changes for extensions. I tried giving it a quick look but it seems like it will take a bit of time to do.

One part I am particularly unsure about is whether the change is backwards compatible. In which case I will have to drop support for gnome 44 and older when I make this change.

**Comment by MissingUser** [comment]
Found this looking for info about the error that may help.

> The settings of extension improved-workspace-indicator@michaelaquilina.github.io had an error:
> ```
> SyntaxError: export declarations may only appear at top level of a module
> ```
> 
> Stack trace:
> ```
> @file:///home/gonzalo/.local/share/gnome-shell/extensions/improved-workspace-indicator@michaelaquilina.github.io/prefs.js:6:16
> _init/GLib.MainLoop.prototype.runAsync/</<@resource:///org/gnome/gjs/modules/core/overrides/GLib.js:266:34
> ```
>

**Comment by MichaelAquilina** [comment]
Yep! There's a general guide here which covers what needs to be changed to get this working on gnome 45 https://gjs.guide/extensions/upgrading/gnome-shell-45.html
